### PR TITLE
Docs: Add a graphical representation of commands executed with yarn start

### PR DESCRIPTION
**Pull request by epeicher**
## Proposed Changes

* Add a section to the document `Tech behind Calypso` with a graphical representation of the commands executed when running `yarn start`

## Testing Instructions

* Apply this patch
* Navigate to `/devdocs/docs/guide/tech-behind-calypso.md`
* Scroll to bottom and check the new section named `Commands executed when you run yarn start`
  CleanShot 2023-11-16 at 16 02 10@2x

* Note: the `mermaid` diagram doesn't work in Calypso docs but it renders when being displayed by Github so the link to GH points to `trunk` and it will only work when merging this PR.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: docs/guide/tech-behind-calypso.md [hunk 1]
```diff
@@ -105,4 +105,14 @@ The way we use Git with Calypso is described in the [Git Workflow document](../g
...
+
+[Link to diagram here](https://github.com/Automattic/wp-calypso/blob/trunk/docs/yarn-start.md)
+
+[Link to the editable diagram here](https://mermaid.live/edit#pako:eNptVE1z0zAQ_Sse9VCHieP4o47jA0wLzNBCgRk4UfcgS3IsqkgaS25qOvnvrK2kJKHWxbP73u5b7a6eEVGUoQKtWqwb7-eVB18pje0F8zaaeDUXojirl_XU2FY9sOIsSZLdf7Dh1DZFqp9KOZxLv8et9IzFrZ0EwdsrnzSMPHgSUniPrDVcyYlLcAnu9_7o2DBB1JrNfpsDX99WLljbSa_quKBHTnPgHdMFhxgguxDmJMaAtZycwog5BQZgOkJ98sEXaC8w3rlDUPZIFTHFm_Mj4PU_oMMZ1kLp-7CCM2kDXge6VQdyDTBvfKmfPEn0TqWnu0qA2BfAZ9_bI1ycUPAqHOJzAr1Q7QPkOSRBEUD78p90sA-yS3kNbrikQ6F7K-Haf1VzKUcxMBt3dxtWaQwNDgKiZM1Xe107-8xZZ0Obob339ztuZYg-vXCXvSBK98Fa0U4wM6YiOgiujfcdUr8D6q2bmYrLcIQOkjpiYbD2rGGOgAj6j4hfj6p5DfFthxBYrjq8YmYomUlcCUbH-zRDf0GS7_slcjIGQuikQ94STSYTtwruoClas3aNOYUNey4ldBvZhq1ZiQr4pazGnbAlKuUWoJ2m2LKPlFvVoqLGwrApwp1VP3pJUGHbju1BHziGhV2_oITClAHpGdlej-vMjYWQrgGDvWsFmBtrtSnCcHDPVtw2XQVNWoeG0wa2qHlcZmEWZzmOE5YtEnyRJJRU0TKv4zSq6WIexRhtt1OksRyiPqEiXs7iOJpneZwv4GlIoinqUREtZvFyvkwv0jzP54tlnAPpj1IgOZrN02yeRuDJ0kWaAYGNNd-6l2h8kMYMv0b8WOL2L2LbjD4)
```
Comment (tyxla) [inline docs/guide/tech-behind-calypso.md]: Would be nice to add some very basic instructions on how to edit the diagram if necessary at some point in the future.

**Review by tyxla — APPROVED** [review]
Really cool addition, thank you 🚀

File: docs/guide/tech-behind-calypso.md (hunk 1 above)
Comment (epeicher) [reply docs/guide/tech-behind-calypso.md]: Good idea, I have added here https://github.com/Automattic/wp-calypso/pull/84250/commits/5f96e963ef134003c77ab95ccff873a6e3d2ab39 some instructions

**Comment by youknowriad** [comment]
To me this feels ill placed in "The technologies behind Calypso". Feels better for its own dedicated markdown page.

**Comment by epeicher** [comment]
> To me this feels ill placed in "The technologies behind Calypso". Feels better for its own dedicated markdown page.

I have created a quick PR to remove it from there https://github.com/Automattic/wp-calypso/pull/103116

**Pull request merged**